### ISSUE TITLE
unsafe-load-any-extension ignored?

### ISSUE THREAD

**Issue by tyralla** [pylint-dev/pylint#7433]
### Bug description

Since 2.15.0,  pylint emits many warnings like the following for [my project](https://github.com/hydpy-dev/hydpy):

```
hydpy\models\dam\dam_model.py:219:58: I1101: Module 'hydpy.cythons.autogen.smoothutils' has no 'smooth_logistic1' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
```

Enabling or disabling the `unsafe-load-any-extension` option does not make a difference. 

Adding the relevant cython extension modules explicitly via the `extension-pkg-allow-list` seems to help (at least silences the warnings).

### Command used

```shell
pylint hydpy
```

### Pylint output

```shell
For example:


hydpy\models\dam\dam_model.py:219:58: I1101: Module 'hydpy.cythons.autogen.smoothutils' has no 'smooth_logistic1' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
```
```


### Expected behavior

no warnings

### Pylint version

```shell
pylint 2.15.2
astroid 2.12.9
python 3.8.10
```

### OS / Environment

Windows (my computer), Linux (Travis-CI)

**Comment by knedlsepp** [comment]
I was able to bisect this issue down to https://github.com/pylint-dev/astroid/commit/bbcc58bd52e7f295b77a8618b19b2364625590a2 which apparently introduced this regression.

**Comment by jacobtylerwalls** [comment]
I suggested rewriting the AstroidManager instance attributes as properties so that they actually impact global state in https://github.com/pylint-dev/astroid/issues/2200#issuecomment-1580325917.
